### ISSUE TITLE
Boxes not displayed correctly in tf model

### ISSUE THREAD

**Issue by dh031200** [ultralytics/ultralytics#1059]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Detection, Export

### Bug

I was converting a custom model to a tflite model for use in the flutter.
The tflite model does not give the correct result value for inference.
The same thing happens in the official model.
There was no change even though I used the updated version by glenn-jocher about 4 hours ago.

### with tflite model:
![image](https://user-images.githubusercontent.com/66017052/220014397-5bdfb7bd-de3d-4cdd-8bfb-27de7d1d63c1.png)

### with pytorch model:
![image](https://user-images.githubusercontent.com/66017052/220014704-4f028e47-b4fc-4c5e-8b9a-7a3c2fd6b374.png)

Below is the execution result shown in the terminal. Nothing seems to be wrong.
```
yolo predict task=detect model=best_face_saved_model/best_face_float32.tflite imgsz=640 source=samples/1.jpg show
Ultralytics YOLOv8.0.41 ? Python-3.9.16 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)
2023-02-20 14:07:39.382515: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
Loading best_face_saved_model/best_face_float32.tflite for TensorFlow Lite inference...
INFO: Created TensorFlow Lite delegate for select TF ops.
2023-02-20 14:07:40.000564: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-02-20 14:07:41.638539: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1613] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 21771 MB memory:  -> device: 0, name: NVIDIA GeForce RTX 3090, pci bus id: 0000:01:00.0, compute capability: 8.6
2023-02-20 14:07:41.638898: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1613] Created device /job:localhost/replica:0/task:0/device:GPU:1 with 12596 MB memory:  -> device: 1, name: NVIDIA GeForce RTX 3090, pci bus id: 0000:02:00.0, compute capability: 8.6
INFO: TfLiteFlexDelegate delegate: 9 nodes delegated out of 246 nodes with 9 partitions.

INFO: Created TensorFlow Lite XNNPACK delegate for CPU.

image 1/1 /home/gpu-1/Project/v8-face/samples/1.jpg: 640x640 4 faces, 132.9ms
Speed: 1.4ms preprocess, 132.9ms inference, 1.6ms postprocess per image at shape (1, 3, 640, 640)
```

### Environment

ubuntu 22.04
Ultralytics YOLOv8.0.41 ? Python-3.10.9 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)

ubuntu 22.04
Ultralytics YOLOv8.0.41 ? Python-3.9.16 torch-1.13.1+cu117 CUDA:0 (NVIDIA GeForce RTX 3090, 24268MiB)

google colab
Ultralytics YOLOv8.0.41 🚀 Python-3.8.10 torch-1.13.1+cu116 CUDA:0 (Tesla T4, 15110MiB)

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@dh031200 thanks for the bug report. This was originally reported in https://github.com/ultralytics/ultralytics/issues/992 and is still an open bug. I can export correctly to TF locally on my Macbook but see errors in CI and in Colab. 

You can see the errors in the new CI benchmarks for example as lower mAP scores on all TF export formats:
https://github.com/ultralytics/ultralytics/actions/runs/4222036985/jobs/7330047057
<img width="1423" alt="Screenshot 2023-02-20 at 01 41 15" src="https://user-images.githubusercontent.com/26833433/220074142-ede6e1ff-9c18-4074-8160-bc28f6d5a996.png">

**Comment by ccclever2** [comment]
i'm having the same issues when exporting to tflite on Linux

![image](https://user-images.githubusercontent.com/55979360/220141338-ad286821-f766-4e16-bab9-0af05a651bb4.png)

**Comment by bmarv** [comment]
+1 having the same problem after exporting on macos. @glenn-jocher is there a previous commit, that exports to tflite correctly?

**Comment by glenn-jocher** [comment]
@bmarv @ccclever2 TF export is a new feature, it’s only been released recently. As mentioned above the onnx2tf package appears to behave differently in different environments. 

When I was developing TF exports everything worked correctly (and still does) on my macbook M2, but as you’re aware and as our CI shows does not in other environments.

If you can help debug this that would help.

TF export for classification models does work BTW, and masks also appear to work correctly for seg models, the issue appears isolated to detection models, probably in the DFL transpose/softmax ops.

**Comment by bmarv** [comment]
the apple silicon requirements.txt would be highly appreciated

**Comment by glenn-jocher** [comment]
@bmarv oh there's no difference in requirements.txt with M1/M2, you just install Python 3.10 from https://www.python.org/ (not 3.11 as torch isn't supported there yet), create a new venv and `pip install ultralytics` followed by `yolo export format=tflite`

**Comment by glenn-jocher** [comment]
@bmarv @ccclever2 CoreML in particular is lightning fast on Apple Silicon. The benchmark results on my 2022 M2 MacBook Air are below. Note the TFLite metrics show correct export also.

```
yolo benchmark imgsz=320
```

<img width="642" alt="Screenshot 2023-02-21 at 20 34 49" src="https://user-images.githubusercontent.com/26833433/220441608-6d3895c8-3f4f-46e0-acb9-a0ad1f17e01e.png">

**Comment by glenn-jocher** [comment]
@bmarv @ccclever2 @dh031200 all TF Saved Model, Protobuf and TFLite modes are now exporting and predicting correctly with ultralytics 8.0.46 via PR https://github.com/ultralytics/ultralytics/pull/1141 as verified in our latest CI benchmarks below. Please update your package with `pip install -U ultralytics` and let us know if this resolves your issue and if you have any other problems. Thank you! 
https://github.com/ultralytics/ultralytics/actions/runs/4271012395/jobs/7435190444

<img width="1139" alt="Screenshot 2023-02-25 at 09 22 15" src="https://user-images.githubusercontent.com/26833433/221370745-43697ce1-0792-43d3-b87b-d4a5fb434eb9.png">

**Comment by ccclever2** [comment]
@glenn-jocher  Nice, I will test it soon, thanks a lot!

**Comment by dh031200** [comment]
@glenn-jocher I will try tomorrow. thanks!

**Comment by hungb2014918** [comment]
@glenn-jocher 
Hi sir, this it's happen again
I convert my model to TFlite on google colab and when it happen when I do **Predict**.
I use version 8.0.181 and got error. I tried down to  8.0.46 but still got the same error

this is my terminal when I convert my model by using `yolo export model="/content/best.pt" format=tflite`

**Ultralytics YOLOv8.0.182 🚀 Python-3.10.12 torch-2.0.1+cu118 CPU (Intel Xeon 2.30GHz)
Model summary (fused): 168 layers, 3008768 parameters, 0 gradients, 8.1 GFLOPs

PyTorch: starting from '/content/best5.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 20, 8400) (5.9 MB)

TensorFlow SavedModel: starting export with tensorflow 2.13.0...

ONNX: starting export with onnx 1.14.1 opset 17...
============= Diagnostic Run torch.onnx.export version 2.0.1+cu118 =============
verbose: False, log level: Level.ERROR
======================= 0 NONE 0 NOTE 0 WARNING 0 ERROR ========================

ONNX: simplifying with onnxsim 0.4.33...
ONNX: export success ✅ 7.2s, saved as '/content/best5.onnx' (11.6 MB)
TensorFlow SavedModel: running 'onnx2tf -i "/content/best5.onnx" -o "/content/best5_saved_model" -nuo --non_verbose'
2023-09-19 18:47:04.926892: E tensorflow/compiler/xla/stream_executor/cuda/cuda_driver.cc:268] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
TensorFlow SavedModel: export success ✅ 26.1s, saved as '/content/best5_saved_model' (29.3 MB)

TensorFlow Lite: starting export with tensorflow 2.13.0...
TensorFlow Lite: export success ✅ 0.0s, saved as '/content/best5_saved_model/best5_float32.tflite' (11.6 MB)

Export complete (29.4s)
Results saved to /content
Predict:         yolo predict task=detect model=/content/best5_saved_model/best5_float32.tflite imgsz=640  
Validate:        yolo val task=detect model=/content/best5_saved_model/best5_float32.tflite imgsz=640 data=/content/drive/MyDrive/TrainNCKH/nckh.yaml  
Visualize:       https://netron.app/
💡 Learn more at https://docs.ultralytics.com/modes/export**
![107](https://github.com/ultralytics/ultralytics/assets/89437524/3418d664-55b8-45d1-ae21-ff2e4cb47e9c)
 
(My English is quite poor, please forgive me)
